### PR TITLE
[woble] Add single BLE connection option. 

### PR DESCRIPTION
**Pull request by turon**
Provides an option to disable WoBLE advertisements while a WoBLE connection is active.

### CODE REVIEW

File: src/adaptations/device-layer/nRF5/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -375,7 +375,11 @@ void BLEManagerImpl::DriveBLEState(void)
     WEAVE_ERROR err = WEAVE_NO_ERROR;
 
     // If the application has enabled WoBLE and BLE advertising...
```
Comment (jaylogue) [inline src/adaptations/device-layer/nRF5/BLEManagerImpl.cpp]: Please update the comment to describe the new behavior.

File: src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h [hunk 2]
```diff
@@ -325,6 +325,16 @@
...
 #endif
 
+/**
+ * WEAVE_DEVICE_CONFIG_SINGLE_WOBLE_CONNECTION
+ *
+ * Limit support for Weave-over-BLE (WoBLE) to a single connection.
+ * When set, WoBLE advertisements will stop while a WoBLE connection is active.
+ */
+#ifndef WEAVE_DEVICE_CONFIG_WOBLE_SINGLE_CONNECTION
+#define WEAVE_DEVICE_CONFIG_WOBLE_SINGLE_CONNECTION 0
```
Comment (jaylogue) [inline src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h]: Can you open an issue to add support for this to the ESP32?

Comment (turon) [reply src/adaptations/device-layer/include/Weave/DeviceLayer/WeaveDeviceConfig.h]: Done: #400

File: src/adaptations/device-layer/nRF5/BLEManagerImpl.cpp (hunk 1 above)
Comment (turon) [reply src/adaptations/device-layer/nRF5/BLEManagerImpl.cpp]: Ok, added to the comment ...
Wasn't sure of the best way to interleave conditional code and extending the comment.
Let me know if you have a better suggestion.

**Pull request merged**